### PR TITLE
migration support for zeebe user tasks

### PR DESCRIPTION
**Pull request by christinaausley**
## Description

Relates to https://github.com/camunda/product-hub/issues/2126. Note that these are changes executing the following:

- [x] Ensure we clearly recommend Zeebe user tasks, and link to migration guide.
- [x] Update [migration guide](https://docs.camunda.io/docs/apis-tools/tasklist-api-rest/migrate-to-zeebe-user-tasks/) to use C8API instead of Zeebe REST API.
- [x] Consider new placement for migration content in APIs/tools in sidebar.
- [x] Explicitly state in migration guide why it is recommended to design a process with one implementation type.

**The following will be added in a separate PR:**

## For 8.6

- [ ] Document that user tasks with Job worker implementation are deprecated (but supported).

## For 8.7

- [ ] Note that the Tasklist API will no longer be supported with 8.9 and recommend migrating.
- [ ] Remove interactive questionnaire in [migration guide](https://docs.camunda.io/docs/apis-tools/tasklist-api-rest/migrate-to-zeebe-user-tasks/)
- [ ] Rename Zeebe user task to Camunda user task and note terminology change with admonition as the XML may still show Zeebe user task.
- [ ] Document that the C8 REST API only supports user tasks with the Camunda user task implementation.
- [ ] Document that embedded forms are supported only for Job worker-based user tasks.
E.g. this section needs to be updated: https://docs.camunda.io/docs/next/components/modeler/web-modeler/advanced-modeling/form-linking/#camunda-form-embedded

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [x] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Comment by christinaausley** [comment]
@aleksander-dytko @volodymyr-melnykc How do you feel about me moving the migration manual to a new section? See [related](https://github.com/camunda/camunda-docs/pull/4557/files).

If this is fine, I will apply it to 8.6 and 8.5 as well.

Note that we still need to change the migration manual to the C8 API instead of the Zeebe REST API in both 8.6 and next.

**Comment by volodymyr-melnykc** [comment]
> @aleksander-dytko @volodymyr-melnykc How do you feel about me moving the migration manual to a new section? See [related](https://github.com/camunda/camunda-docs/pull/4557/files).

@christinaausley Moving the migration manual to a new dedicated section under the "APIs & Tools" makes sense to me.

**Comment by christinaausley** [comment]
@volodymyr-melnykc I have started updating the endpoints in the Zeebe user task guide for the C8 API, but could you please make any final adjustments to the technical details you would like to make and I will backport?

**Comment by christinaausley** [comment]
Note that @volodymyr-melnykc and I are syncing this week on plans for this and https://github.com/camunda/product-hub/issues/2126.

File: versioned_docs/version-8.6/apis-tools/tasklist-api-rest/tasklist-api-rest-overview.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ sidebar_position: 1
 description: "Build applications for human-centered processes by querying user tasks, assigning users, and completing tasks with the Tasklist API."
 ---
 
+:::note
+Camunda introduced the [Camunda 8 REST API](/apis-tools/camunda-api-rest/camunda-api-rest-overview.md) with `8.6` to manage and query process entities such as processes, decisions, forms, and user tasks.
```
Comment (christinaausley) [inline versioned_docs/version-8.6/apis-tools/tasklist-api-rest/tasklist-api-rest-overview.md:9]: Is this still accurate @volodymyr-melnykc?

File: versioned_docs/version-8.5/apis-tools/tasklist-api-rest/tasklist-api-rest-overview.md [hunk 2]
```diff
@@ -5,6 +5,12 @@ sidebar_position: 1
 description: "Build applications for human-centered processes by querying human tasks, assigning users, and completing tasks with the Tasklist API."
 ---
 
+:::note
```
Comment (christinaausley) [inline versioned_docs/version-8.5/apis-tools/tasklist-api-rest/tasklist-api-rest-overview.md:8]: Similar comment to double check here @volodymyr-melnykc.

**Comment by christinaausley** [comment]
@volodymyr-melnykc If [this](https://github.com/camunda/product-hub/issues/2126#issuecomment-2498468491) is accurate, this PR just requires your review and a few things to check for accuracy.

Could you please ensure the migration guide (changed from Zeebe REST to C8) is up to date and accurate?

> Explicitly state in migration guide why it is recommended to design a process with one implementation type.

How would you like this to look in the documentation?

**Comment by akeller** [comment]
@christinaausley will this be delivered with alpha2? Or still largely WIP? I'm trying to get a handle on when we can move forward with the API removal PRs (Zeebe REST API in this case).

**Comment by christinaausley** [comment]
This can be merged as soon as @volodymyr-melnykc has a look -- he mentioned he will have time before the end of this week, so I can ensure the API removals also go out with the alpha next week. @akeller

**Comment by volodymyr-melnykc** [comment]
I'm reviewing it currently.
Regarding the delivery date for these docs. The related [product epic](https://github.com/camunda/product-hub/issues/2126) has been moved to alpha3. Should we also aim to deliver the docs in alpha3?
@christinaausley @akeller

**Comment by volodymyr-melnykc** [comment]
@christinaausley I reviewed the changes (they looked good!) and updated several more details:
- The task application architecture doc
- The migration guide to the Camunda (Zeebe) user task
- Renamed "Zeebe user task" and "Camunda user task"

Please review!

The doc for user task migration is called "migrate-to-zeeb-user-task.md". Since the implementation type is renamed to "Camunda user task", should we update the name of the file as well?
 

cc @akeller

**Comment by akeller** [comment]
> I'm reviewing it currently. Regarding the delivery date for these docs. The related [product epic](https://github.com/camunda/product-hub/issues/2126) has been moved to alpha3. Should we also aim to deliver the docs in alpha3? @christinaausley @akeller

@volodymyr-melnykc, just confirming, there will be no end user changes that we should document in this context for alpha2?

I would love to see if get these docs in a good place to merge ASAP, or in this case, shortly after alpha2 if possible. We have many changes coming in alpha3 and the race to get things done for 8.7. If there is a way we can ship these docs changes incrementally to keep things moving, let's do that.

**Comment by christinaausley** [comment]
Perhaps I did not communicate clearly enough in the PR description, but this PR is solely for changes with 8.5 and 8.6, which are changes that can likely already be merged in. I have plans to adjust all of the required changes in `next` in a separate PR.

With this in mind, could you please review my changes to the versioned docs @volodymyr-melnykc? I will sift your additional changes in `next` into a separate PR.

**Comment by christinaausley** [comment]
Changes specifically for Zeebe to Camunda user tasks will be covered in https://github.com/camunda/camunda-docs/pull/4723, alongside the following:

## For 8.6

- [ ] Document that user tasks with Job worker implementation are deprecated (but supported).

## For 8.7

- [ ] Note that the Tasklist API will no longer be supported with 8.9 and recommend migrating.
- [ ] Remove interactive questionnaire in [migration guide](https://docs.camunda.io/docs/apis-tools/tasklist-api-rest/migrate-to-zeebe-user-tasks/)
- [ ] Rename Zeebe user task to Camunda user task and note terminology change with admonition as the XML may still show Zeebe user task.
- [ ] Document that the C8 REST API only supports user tasks with the Camunda user task implementation.
- [ ] Document that embedded forms are supported only for Job worker-based user tasks.
E.g. this section needs to be updated: https://docs.camunda.io/docs/next/components/modeler/web-modeler/advanced-modeling/form-linking/#camunda-form-embedded

@volodymyr-melnykc Please review this PR -- my expectation is that all of these changes can go live and be merged now? Changes for `next` are still a WIP.

Lastly, can your changes to the specifications be merged and live as well? Is this work already complete? If so, should this also be backported?

Thank you for your assistance and apologies for the back and forth on this.

**Comment by volodymyr-melnykc** [comment]
@christinaausley Your changes look good and we can merge!

The changes that I made are related to the ongoing updates in the product epic (https://github.com/camunda/product-hub/issues/2126).  They're for the "next". I don't see risks in merging them as well, though there will be more follow-ups (which we can do later).

**Review by volodymyr-melnykc — APPROVED** [review]

File: versioned_docs/version-8.5/apis-tools/tasklist-api-rest/migrate-to-zeebe-user-tasks.md [hunk 3]
```diff
@@ -12,7 +12,6 @@ import TableTextSmall from "./assets/react-components/TableTextSmall";
 import userTaskMigrationDecisionHelperForm from "./assets/forms/userTaskMigrationDecisionHelperForm.js";
 import "./assets/css/condensedTable.module.css";
 import styles from "./assets/css/cleanImages.module.css";
-import APIArchitectureImg from './assets/img/api-architecture.png';
```
Comment (pepopowitz) [inline versioned_docs/version-8.5/apis-tools/tasklist-api-rest/migrate-to-zeebe-user-tasks.md:15]: @christinaausley this line removal is causing your build error. This image is referenced in the doc, and can't be removed.

Comment (christinaausley) [reply versioned_docs/version-8.5/apis-tools/tasklist-api-rest/migrate-to-zeebe-user-tasks.md:15]: THANK YOU!

File: docs/apis-tools/migration-manuals/migrate-to-zeebe-user-tasks.md [hunk 4]
```diff
@@ -181,176 +167,183 @@ We recommend you migrate process-by-process, allowing you to thoroughly test the
 
 Repeat these steps for all user tasks in the process. Then, deploy the process to your development cluster and test it by running the process and ensuring your custom task applications work.
 
-## Use the new Zeebe Task API
+## Use Camunda 8 API
 
 :::note
 The Tasklist REST API is not deprecated, and you still need it for queries on both task types.
```
Comment (akeller) [inline docs/apis-tools/migration-manuals/migrate-to-zeebe-user-tasks.md:173]: This note won't be quite right in 8.7. The Tasklist REST API will be marked as deprecated, but not removed until 8.9. Will this be adjusted in a future PR? Internal context [here](https://docs.google.com/document/d/1aqwSXP8uF3I7cV6Rz-HmBYIXSltwlmfjHKmmlVWBg6o/edit?tab=t.0#heading=h.zhyuwlliwpqj).

**Review by akeller — APPROVED** [review]
I left one non-blocking comment (as we can address the Tasklist REST API note in a future PR).

File: docs/apis-tools/migration-manuals/migrate-to-zeebe-user-tasks.md (hunk 4 above)
Comment (christinaausley) [reply docs/apis-tools/migration-manuals/migrate-to-zeebe-user-tasks.md:173]: To adjust with https://github.com/camunda/camunda-docs/pull/4723 👍

**Pull request merged**